File: java/org.financer.client.javafx/src/test/java/org/financer/client/javafx/main/ProfileTest.java
package org.financer.client.javafx.main;

import com.jfoenix.controls.JFXButton;
import com.jfoenix.controls.JFXPasswordField;
import com.jfoenix.controls.JFXTextField;
import javafx.scene.control.Button;
import javafx.scene.control.Hyperlink;
import javafx.scene.control.Label;
import org.financer.client.domain.model.user.User;
import org.financer.client.javafx.local.LocalStorageImpl;
import org.financer.util.Hash;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;

@SuppressWarnings("WeakerAccess")
@Tag("integration")
public class ProfileTest extends AbstractFinancerApplicationTest {

    @BeforeEach
    public void setUpEach() throws Exception {
        super.setUpEach();
    }

    @Test
    public void testChangePassword() {
        final String newPassword = "newPassword";
        register(user(), password());

        clickOn((Button) find("#profileTabBtn"));
        sleep(SHORT_SLEEP);
        clickOn((Hyperlink) find("#changePasswordLink"));

        clickOn((JFXPasswordField) find("#oldPasswordField"));
        write(password());
        clickOn((JFXPasswordField) find("#newPasswordField"));
        write(newPassword);
        clickOn((JFXPasswordField) find("#repeatNewPasswordField"));
        write(newPassword);

        confirmDialog();

        sleep(SHORT_SLEEP);

        Assertions.assertEquals(Hash.create(newPassword, ((User) LocalStorageImpl.getInstance().readObject("user")).getPassword().getSalt()),
                ((User) LocalStorageImpl.getInstance().readObject("user")).getPassword().getHashedPassword());
    }

    @Test
    public void testChangePersonalInformation() {
        final String newName = "Peter";
        register(user(), password());

        clickOn((Button) find("#profileTabBtn"));
        sleep(SHORT_SLEEP);
        clickOn((JFXButton) find("#editPersonalInformationBtn"));
        sleep(SHORT_SLEEP);
        clickOn((JFXTextField) find("#nameField"));
        eraseText(formatter.format(user().getName()).length());
        write(newName);
        confirmDialog();
        sleep(SHORT_SLEEP);

        Assertions.assertEquals(newName, ((User) LocalStorageImpl.getInstance().readObject("user")).getName().getFirstName());
        Assertions.assertEquals(newName + " " + user().getName().getSurname(), ((Label) find("#fullNameLabel")).getText());
    }
}

File: java/org.financer.client.javafx/src/test/kotlin/org/financer/client/javafx/main/ProfileTest.kt
package org.financer.client.javafx.main

import com.jfoenix.controls.JFXButton
import com.jfoenix.controls.JFXPasswordField
import com.jfoenix.controls.JFXTextField
import javafx.scene.Node
import javafx.scene.control.Button
import javafx.scene.control.Hyperlink
import javafx.scene.control.Label
import org.financer.client.domain.model.user.User
import org.financer.client.javafx.local.LocalStorageImpl
import org.financer.util.Hash
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import java.io.Serializable

@Tag("integration")
class ProfileTest : AbstractFinancerApplicationTest() {
    @BeforeEach
    @Throws(Exception::class)
    public override fun setUpEach() {
        super.setUpEach()
    }

    @Test
    fun testChangePassword() {
        val newPassword = "newPassword"
        register(user(), password())
        clickOn(find<Node>("#profileTabBtn") as Button)
        sleep(AbstractFinancerApplicationTest.Companion.SHORT_SLEEP.toLong())
        clickOn(find<Node>("#changePasswordLink") as Hyperlink)
        clickOn(find<Node>("#oldPasswordField") as JFXPasswordField)
        write(password())
        clickOn(find<Node>("#newPasswordField") as JFXPasswordField)
        write(newPassword)
        clickOn(find<Node>("#repeatNewPasswordField") as JFXPasswordField)
        write(newPassword)
        confirmDialog()
        sleep(AbstractFinancerApplicationTest.Companion.SHORT_SLEEP.toLong())
        Assertions.assertEquals(Hash.create(newPassword, (LocalStorageImpl.getInstance().readObject<Serializable>("user") as User?)!!.password!!.salt),
                (LocalStorageImpl.getInstance().readObject<Serializable>("user") as User?)!!.password!!.hashedPassword)
    }

    @Test
    fun testChangePersonalInformation() {
        val newName = "Peter"
        register(user(), password())
        clickOn(find<Node>("#profileTabBtn") as Button)
        sleep(AbstractFinancerApplicationTest.Companion.SHORT_SLEEP.toLong())
        clickOn(find<Node>("#editPersonalInformationBtn") as JFXButton)
        sleep(AbstractFinancerApplicationTest.Companion.SHORT_SLEEP.toLong())
        clickOn(find<Node>("#nameField") as JFXTextField)
        eraseText(formatter!!.format(user().name!!)!!.length)
        write(newName)
        confirmDialog()
        sleep(AbstractFinancerApplicationTest.Companion.SHORT_SLEEP.toLong())
        Assertions.assertEquals(newName, (LocalStorageImpl.getInstance().readObject<Serializable>("user") as User?)!!.name!!.firstName)
        Assertions.assertEquals(newName + " " + user().name!!.surname, (find<Node>("#fullNameLabel") as Label).text)
    }
}